index.bs

```diff
@@ -1710,3 +1738,161 @@ When [=a browsing context is discarded=], run the following steps:
+
+#### entryAdded #### {#event-log-entryAdded}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        LogEntryAddedEvent = {
+         method: "log.entryAdded",
+         params: LogEntry
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for log.entryAdded">
+
+When the [=Printer=] algorithm is invoked with arguments |logLevel|, |args| and,
+optionally, <var ignore>options</var>:
+
+1. If |logLevel| is "<code>error</code>" or "<code>assert</code>", let |level|
+   be "<code>error</code>". If |logLevel| is "<code>debug</code>" or
+   "<code>trace</code>" let |level| be "<code>debug</code>". If |logLevel| is
+   "<code>warn</code>", let |level| be "<code>warning</code>". Otherwise let
+   |level| be "<code>info</code>".
+
+1. Let |timestamp| be a [=time value=] representing the current date and time in UTC.
+
+1. Let |text| be an implementation-defined string representing the arguments
+   passed as |args|, or null if there is no appropriate string representation.
+
+1. Let |serialized args| be a new list.
+
+1. For each |arg| of |args|, append the result of [=serialize as a remote
+   value=] given |arg| to |serialized args|.
+
+1. Let |realm| be the [=realm id=] of the [=current Realm Record=].
+
+1. Let |extra| be a map matching the <code>ConsoleLogExtra</code> production,
+   with the <code>type</code> field set to |logLevel|, the <code>real</code>
+   field set to |realm| and the <code>args</code> field set to |serialized
+   args|.
+
+1. Let |entry| be a map matching the <code>LogEntry</code> production, with the
+   <code>type</code> field set to "<code>console</code>", the <code>level</code>
+   field set to |level|, the <code>text</code> field set to |text|, the
+   <code>timestap</code> field set to |timestamp| and the <code>extra</code>
+   field set to |extra|.
+
+1. Let |body| be a map matching the <code>LogEntryAdded</code> production, with
+   the <code>params</code> field set to |entry|.
+
+1. Let |related contexts| be an empty set.
+
+1. Let |settings| be the [=current settings object=]
+
+1. Let |related contexts| be the result of [=get related contexts=] given |settings|.
+
+1. [=Emit an event=] with |body| and |related contexts|.
+
+After an implementation is required to [=report an exception=] from |script|,
+and the error is [=not handled=]:
+
+1. Let |settings| be |script|'s [=script/settings object=].
+
+1. Let |message| be the implementation-defined string that was
+   used as the message in [=report an error=].
+
+1. Let |line number| and |column number| be the one-based line number and
+   zero-based column number, respectively, in the resource containing |script|,
+   at the line where the error originated.
+
+1. Let |url| be the result of running the [=URL serializer=], given
+   |settings|'s [=creation URL=].
+
+1. Let |extra| a map matching the <code>JavascriptLogExtra</code> production,
```

Comment:
How about other kinds of errors that also show up in the console, or say deprecation messages? Will we need to write additional prose to handle those cases?

Reply:
Yes, I think basically every type needs to be handled specially. It's a bit annoying but we can probably have a single set of steps for all the ones which are basically implementation-defined just explaining which types to use for the various common kinds of message.